### ISSUE TITLE
cppcheck missing "constParameter" rule

### ISSUE THREAD

**Issue by VirtualTim** [SonarOpenCommunity/sonar-cxx#1912]
### Description

Sonarqube cppcheck is missing the "constParameter" rule.

Pretty much the same as #1636, but with constParameter instead of nullPointerArithmeticRedundantCheck.

**Comment by guwirth** [comment]
Hi @VirtualTim, thanks for the hint. Which version of cppcheck are you using? Regards

**Comment by VirtualTim** [comment]
Sorry, I should have included that information. I'm using cppcheck version 1.90.
Exert from the report:
```
<results version="2">
  <cppcheck version="1.90"/>
  <errors>
    <error id="constParameter" severity="style" msg="Parameter 'rect' can be declared with const" verbose="Parameter 'rect' can be declared with const" cwe="398">
      <location file="/.../someFileh" line="137" column="42"/>
    </error>
  </errors>
</results>
```

**Comment by guwirth** [comment]
Hi @VirtualTim,

you are right, the rules are listed in https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/cxx-sensors/src/main/resources/cppcheck.xml and `constParameter` is not in the list. We are using `generate_cppcheck_resources` https://github.com/SonarOpenCommunity/sonar-cxx/tree/master/cxx-sensors/src/tools to create the list.

If you are calling `cppcheck.exe --errorlist --xml-version=2 > cppcheck-errorlist.xml` do you get the rule?

Regards,

**Comment by VirtualTim** [comment]
Yeah you're right, it's not listed in `cppcheck.exe --errorlist`. I tested 1.9 and 2.1. I'll log a bug on cppcheck.
